### PR TITLE
Implement load jobs

### PR DESCRIPTION
**Pull request by jskeet**
Split into separate commits for ease of review, as this PR fixes a couple of other issues at the same time.

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+using Google.Apis.Bigquery.v2.Data;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.BigQuery.V2
+{
+    /// <summary>
+    /// Options for <c>CreateExtractJob</c> operations.
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs]: Create\*Load\*Job

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        /// Datastore backup, an invalid error is returned in the job result.
+        /// </summary>
+        public IList<string> ProjectionFields { get; set; }
+
+        /// <summary>
+        /// The format of the source files.
+        /// </summary>
+        public FileFormat SourceFormat { get; set; }
+
+        // All: schema, destination, sourceURIs
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs]: Is this line meant to be here?

Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs]: Nope :( Thanks, will fix this along with the other silly typos.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+        /// <summary>
+        /// Specifies a string that represents a null value in a CSV file. For example,
+        /// if you specify "\N", BigQuery interprets "\N" as a null value when loading
+        /// a CSV file. The default value is the empty string. If you set this property
+        /// to a custom value, BigQuery throws an error if an empty string is present for
+        /// all data types except for STRING and BYTE. For STRING and BYTE columns,
+        /// BigQuery interprets the empty string as an empty value.
+        /// </summary>
+        public string NullMarker { get; set; }
```
Comment (evildour) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs]: Look unused. Should this be copied in `ModifyRequest`?

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        /// sensitive and must be top-level properties. If no properties are specified,
+        /// BigQuery loads all properties. If any named property isn't found in the Cloud
+        /// Datastore backup, an invalid error is returned in the job result.
+        /// </summary>
+        public IList<string> ProjectionFields { get; set; }
+
+        /// <summary>
+        /// The format of the source files.
+        /// </summary>
+        public FileFormat SourceFormat { get; set; }
```
Comment (evildour) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs]: Same here

Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs]: Thanks, will fix both of those in a minute.

**Pull request merged**